compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9857,7 +9857,11 @@ private STNode parsePipeToken() {
     }
 
     private boolean isTypeStartingToken(SyntaxKind nodeKind) {
-        switch (nodeKind) {
+        return isTypeStartingToken(nodeKind, getNextNextToken());
+    }
+    
+    static boolean isTypeStartingToken(SyntaxKind nextTokenKind, STToken nextNextToken) {
```

Comment:
```suggestion
    private static boolean isTypeStartingToken(SyntaxKind nextTokenKind, STToken nextNextToken) {
```

Reply:
This is call by `isEndOfLetVarDeclarations` which is a static method. Hence `isTypeStartingToken` has to be static